### PR TITLE
Bump submariner-operator version and adjust to changes

### PR DESCRIPTION
**Pull request by tpantelis**

### CODE REVIEW

**Review by dfarrell07 — APPROVED** [review]

File: cmd/subctl/join.go [hunk 1]
```diff
@@ -67,10 +67,10 @@ var joinCmd = &cobra.Command{
 		clientProducer, err := client.NewProducerFromRestConfig(clientConfig.Config)
 		exit.OnError(status.Error(err, "Error creating the client producer"))
 
-		client, err := controllerClient.New(clientConfig.Config, controllerClient.Options{})
+		client, err := controllerclient.New(clientConfig.Config, controllerclient.Options{})
```
Comment (skitt) [inline cmd/subctl/join.go:70]: Can’t we use the general client from `clientProducer` here?

File: cmd/subctl/join.go [hunk 2]
```diff
@@ -291,11 +291,10 @@ func determineClusterID(status reporter.Interface) {
 	}
 }
 
-func getNetworkDetails(clientProducer client.Producer, status reporter.Interface) *network.ClusterNetwork {
+func getNetworkDetails(controllerClient controllerclient.Client, status reporter.Interface) *network.ClusterNetwork {
 	status.Start("Discovering network details")
 
-	networkDetails, err := network.Discover(clientProducer.ForDynamic(), clientProducer.ForKubernetes(), clientProducer.ForOperator(),
-		constants.OperatorNamespace)
+	networkDetails, err := network.Discover(controllerClient, constants.OperatorNamespace)
```
Comment (skitt) [inline cmd/subctl/join.go:297]: Likewise, the function declaration could be left unchanged and `clientProducer.ForGeneral()` used instead.

**Review by skitt — CHANGES_REQUESTED** [review]

File: cmd/subctl/join.go (hunk 1 above)
Comment (tpantelis) [reply cmd/subctl/join.go:70]: The `clientProducer` created on L67 is still the **submariner-operator** version. I'd have to migrate all the `join` code to use the new `clientProducer` but this PR is intended to adjust to the **submariner-operator** changes so wanted to keep the changes here to a minimum.

File: cmd/subctl/join.go (hunk 2 above)
Comment (tpantelis) [reply cmd/subctl/join.go:297]: See above comment.

File: cmd/subctl/join.go (hunk 1 above)
Comment (skitt) [reply cmd/subctl/join.go:70]: Ah right, I see!

**Review by skitt — APPROVED** [review]

**Pull request merged**